src/Factory/AbstractNonceFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace RebelCode\WordPress\Nonce\Factory;
+
+use RebelCode\WordPress\Nonce\NonceInterface;
+
+/**
+ * Basic functionality for a nonce factory.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractNonceFactory
+{
+    /**
+     * Creates a new nonce instance.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $id   The ID of the nonce to be generated.
+     * @param array  $args Optional additional arguments.
+     *
+     * @return NonceInterface
+     */
+    protected function _make($id, $args = [])
+    {
+        $code = $this->_generateNonceCode($id);
```

Comment:
The nonce ID is the action context that is used by WordPress.

I didn't really consider the optional nature of the action context, mostly because WordPress would generate the same code if asked to do so multiple times without an action context. This is also true for identical contexts.

But regardless, that's not a reason to limit things. Do you think it should optional?